packages/lesswrong/components/posts/UserSuggestNominations.tsx

```diff
@@ -1,33 +1,131 @@
 import React from 'react';
-import { useLocation } from '../../lib/routeUtil';
+import { useLocation, useNavigate } from '../../lib/routeUtil';
 import { registerComponent, Components } from '../../lib/vulcan-lib';
 import { useCurrentUser } from '../common/withUser';
+import Tabs from '@material-ui/core/Tabs';
+import Tab from '@material-ui/core/Tab';
+import {AnalyticsContext} from '@/lib/analyticsEvents'
+import {preferredHeadingCase} from '@/themes/forumTheme'
+import withErrorBoundary from '@/components/common/withErrorBoundary'
+import moment from 'moment'
 
-const UserSuggestNominations = () => {
-    const { SectionTitle, SingleColumnSection, ErrorBoundary, PostsByVoteWrapper } = Components
-    const currentUser = useCurrentUser()
-
-    const { params } = useLocation();
-    
-    // Handle url-encoded special case, otherwise parseInt year
-    const year = ['≤2020', '%e2%89%a42020'].includes(params?.year) ?
-      '≤2020' :
-      parseInt(params?.year)
-    if (!currentUser) return null
-
-    return <ErrorBoundary>
-      <SingleColumnSection>
-        <SectionTitle title={`Your Strong Upvotes from ${year}`}/>
+const styles = (theme: ThemeType): JssStyles => ({
+  headline: {
+    color: theme.palette.grey[1000],
+    marginBottom: 20,
+    [theme.breakpoints.down('sm')]: {
+      marginTop: 20,
+      marginBottom: 10,
+    }
+  },
+  tab: {
+    fontSize: 14,
+    [theme.breakpoints.down('xs')]: {
+      fontSize: 13,
+    }
+  }
+});
+
+
+const dateStr = (startDate?: Date) =>
+  startDate ? moment(startDate).format('YYYY-MM-DD') : ''
+
+const UserSuggestNominations = ({classes}: { classes: ClassesType }) => {
+  const currentUser = useCurrentUser()
+  const navigate = useNavigate()
+  const {params, location, query} = useLocation()
+  
+  const {
+    SectionTitle,
+    SingleColumnSection,
+    ErrorBoundary,
+    PostsByVoteWrapper,
+    ReadHistoryTab,
+    PostsListUserCommentedOn,
+    Typography,
+    LWTooltip,
+    AllPostsPage
+  } = Components
+
+  // Handle url-encoded special case, otherwise parseInt year
+  const before2020 = '≤2020'
+  const year = [before2020, '%e2%89%a42020'].includes(params?.year) ?
+    before2020 :
+    parseInt(params?.year)
+  if (!currentUser) return null
+
+  const startDate = year === before2020 ? undefined : new Date(year, 0, 1)
+  const endDate = year === before2020 ? new Date(2020, 0, 1) : new Date(year + 1, 0, 1)
+
+  const handleChangeTab = (e: React.ChangeEvent, value: string) => {
+    navigate({
+      ...location,
+      search: `?tab=${value}` + 
+        (value === 'all' ? `&after=${(dateStr(startDate))}&before=${dateStr(endDate)}&timeframe=yearly` : ''),
```

Comment:
This should probably be using qs.stringify:
```
const newQuery = {
  tab: value,
  ...(value === 'all' && { after: dateStr(startDate), before: dateStr(endDate), timeframe: 'yearly' })
};

navigate({
  ...location,
  search: `?${qs.stringify(newQuery)`
});
```

Reply:
nice!